src/core/thread/types.d

```diff
@@ -0,0 +1,79 @@
+/**
+ * This module provides types and constants used in thread package.
+ *
+ * Copyright: Copyright Sean Kelly 2005 - 2012.
+ * License: Distributed under the
+ *      $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0).
+ *    (See accompanying file LICENSE)
+ * Authors:   Sean Kelly, Walter Bright, Alex Rønne Petersen, Martin Nowak
+ * Source:    $(DRUNTIMESRC core/thread/osthread.d)
+ */
+
+module core.thread.types;
+
+/**
+ * Represents the ID of a thread, as returned by $(D Thread.)$(LREF id).
+ * The exact type varies from platform to platform.
+ */
+version (Windows)
+    alias ThreadID = uint;
+else
+version (Posix)
+{
+    import core.sys.posix.pthread;
+
+    alias ThreadID = pthread_t;
+}
+
+struct ll_ThreadData
+{
+    ThreadID tid;
+    version (Windows)
+        void delegate() nothrow cbDllUnload;
+}
+
+version (GNU)
+{
+    import gcc.builtins;
+
+    version (GNU_StackGrowsDown)
+        enum isStackGrowsDown = true;
```

Comment:
This was renamed, not for the better.

Reply:
I renamed it because its "type" changed: `bool enum` is used instead of `version` because it is need to pass it between modules.

Reply:
Okay, but then please change it to `isStackGrowingDown` or so.